x/nexus/keeper/msg_server.go

```diff
@@ -204,21 +210,40 @@ func (s msgServer) deactivateChain(ctx sdk.Context, chain exported.Chain) {
 	)
 }
 
-func isActivationThresholdMet(ctx sdk.Context, nexus types.Nexus, staking types.StakingKeeper, chain exported.Chain) bool {
-	sumConsensusPower := sdk.ZeroInt()
-	maintainers := nexus.GetChainMaintainers(ctx, chain)
+func isActivationThresholdMet(ctx sdk.Context, nexus types.Nexus, slashing types.SlashingKeeper, snapshotter types.Snapshotter, chain exported.Chain) error {
+	isTombstoned := func(v snapshot.ValidatorI) bool {
+		consAdd, err := v.GetConsAddr()
+		if err != nil {
+			return true
+		}
 
-	for _, maintainer := range maintainers {
-		validator := staking.Validator(ctx, maintainer)
+		return slashing.IsTombstoned(ctx, consAdd)
+	}
 
-		if validator == nil || !validator.IsBonded() || validator.IsJailed() {
-			continue
-		}
+	isProxyActive := func(v snapshot.ValidatorI) bool {
+		_, isActive := snapshotter.GetProxy(ctx, v.GetOperator())
 
-		sumConsensusPower = sumConsensusPower.AddRaw(validator.GetConsensusPower(staking.PowerReduction(ctx)))
+		return isActive
 	}
 
-	return nexus.GetParams(ctx).ChainActivationThreshold.IsMet(sumConsensusPower, staking.GetLastTotalPower(ctx))
+	filter := funcs.And(
+		snapshot.ValidatorI.IsBonded,
+		funcs.Not(snapshot.ValidatorI.IsJailed),
+		funcs.Not(isTombstoned),
+		isProxyActive,
+	)
+
+	params := nexus.GetParams(ctx)
+
+	_, err := snapshotter.CreateSnapshot(
+		ctx,
+		nexus.GetChainMaintainers(ctx, chain),
+		filter,
+		snapshot.QuadraticWeightFunc,
+		params.ChainActivationThreshold,
+	)
+
+	return err
```

Comment:
```suggestion
	return err == nil
```
I really don't like an `IsXXX` function to return err

Reply:
Made it a method of msg server, so I can log the error properly, and also keeps the check below clean as before.